x-pack/plugins/alerting/server/alerts_client/types.ts

```diff
@@ -96,17 +96,22 @@ export interface TrackedAlerts<
   recovered: Record<string, LegacyAlert<State, Context>>;
 }
 
+// allows Partial on nested objects
+export type RecursivePartial<T> = {
```

Comment:
Not sure if this is still needed, but for some reason I extended this pattern a bit in event log to handle arrays:

https://github.com/elastic/kibana/blob/fd1bad1f66be58a716201322f2e5a1a84f4ae458/x-pack/plugins/event_log/generated/schemas.ts#L19-L21

Reply:
Turns out they're everywhere! There's a `DeepPartial` in `@kbn/utility-types` that I ended up using. Updated in https://github.com/elastic/kibana/pull/160466/commits/425c34f78e5632eea4cbce5c3e5c58712b34df8a